Unable to find packages.json on Windows Manual Install

**Describe the bug**

I grabbed the latest changeset, d7ed71e, on the develop branch. I think issue #4787 has been resolved, but I'm getting an error that packages.json cannot be found.

Relevant snippet:

```
[ERROR] npm - npm ERR! error in C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\node_modules\ep_etherpad-lite: ENOENT: no such file or directory, open 'C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\node_modules\ep_etherpad-lite\package.json'
```

The file **ep_etherpad-lite** exists and the contents are "../src". If use a file explorer to navigate up one level and into **/src**, I find **packages.json**. I don't think Windows supports using a file for path redirection this way. I tried replacing **ep_etherpad-lite** with a shortcut, but that didn't work. I deleted **ep_etherpad-lite**, created a folder in its place and copied **packages.json** into this folder: that caused this issue to go away and **start.bat** seems to run successfully. 

However, **start.bat** auto exits and returns to the command line and there doesn't appear to be a server running in the background. I don't know if that is a side effect of the change I described above. I then tried using **node src/node/server.js** to start the server instead, but auto-quits in the same way. 

1) Here's the complete log from before I started playing with the **ep_etherpad-lite** file.

```
C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop>start.bat

C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop>node src\node\server.js
[2021-02-16 19:00:20.780] [DEBUG] console - Running on Node v14.15.5 (minimum required Node version: 10.17.0)
[2021-02-16 19:00:20.793] [INFO] console - All relative paths will be interpreted relative to the identified Etherpad base dir: C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop
[2021-02-16 19:00:20.794] [INFO] console - Random string used for versioning assets: 04155d40
[2021-02-16 19:00:20.795] [DEBUG] AbsolutePaths - Relative path "settings.json" can be rewritten to "C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\settings.json"
[2021-02-16 19:00:20.796] [DEBUG] AbsolutePaths - Relative path "credentials.json" can be rewritten to "C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\credentials.json"
[2021-02-16 19:00:20.797] [WARN] console - No settings file found in C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\settings.json. Continuing using defaults!
[2021-02-16 19:00:20.797] [INFO] console - No credentials file found in C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\credentials.json. Ignoring.
[2021-02-16 19:00:20.798] [WARN] console - No "skinName" parameter found. Please check out settings.json.template and update your settings.json. Falling back to the default "colibris".
[2021-02-16 19:00:20.799] [INFO] console - Using skin "colibris" in dir: C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\src\static\skins\colibris
[2021-02-16 19:00:20.799] [INFO] console - Session key file "C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\SESSIONKEY.txt" not found. Creating with random contents.
[2021-02-16 19:00:20.800] [WARN] console - DirtyDB is used. This is not recommended for production. File location: C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\var\dirty.db
[2021-02-16 19:00:21.142] [INFO] server - Starting Etherpad...
[2021-02-16 19:00:21.180] [INFO] runNpm - Executing command: npm ls --long --json --depth=0
[2021-02-16 19:00:21.330] [INFO] runNpm - npm --version: 6.14.11
[2021-02-16 19:00:21.615] [ERROR] npm - npm ERR! error in C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\node_modules\ep_etherpad-lite: ENOENT: no such file or directory, open 'C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\node_modules\ep_etherpad-lite\package.json'
[2021-02-16 19:00:21.623] [ERROR] runNpm - npm command failed: npm ls --long --json --depth=0
[2021-02-16 19:00:21.624] [ERROR] server - Error occurred while starting Etherpad
[2021-02-16 19:00:21.625] [ERROR] server - Error: Command exited non-zero: npm ls --long --json --depth=0
    at exports (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\src\node\utils\run_cmd.js:71:25)
    at exports (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\src\node\utils\run_npm.js:25:13)
    at Object.exports.getPackages (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\src\static\js\pluginfw\plugins.js:73:14)
    at Object.exports.update (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\src\static\js\pluginfw\plugins.js:56:34)
    at Object.exports.start (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\src\node\server.js:137:19)
[2021-02-16 19:00:21.625] [INFO] server - Exiting...
[2021-02-16 19:00:21.626] [INFO] server - Waiting for Node.js to exit...
```

2) Log from after I deleted the **ep_etherpad-lite** file and copied **packages.json** into a new folder in its place.

```
C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop>node ./src/node/server.js
[2021-02-16 19:34:11.334] [DEBUG] console - Running on Node v12.20.2 (minimum required Node version: 10.17.0)
[2021-02-16 19:34:11.348] [INFO] console - All relative paths will be interpreted relative to the identified Etherpad base dir: C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop
[2021-02-16 19:34:11.349] [INFO] console - Random string used for versioning assets: 5a4d5cee
[2021-02-16 19:34:11.349] [DEBUG] AbsolutePaths - Relative path "settings.json" can be rewritten to "C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\settings.json"
[2021-02-16 19:34:11.350] [DEBUG] AbsolutePaths - Relative path "credentials.json" can be rewritten to "C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\credentials.json"
[2021-02-16 19:34:11.350] [WARN] console - No settings file found in C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\settings.json. Continuing using defaults!
[2021-02-16 19:34:11.351] [INFO] console - No credentials file found in C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\credentials.json. Ignoring.
[2021-02-16 19:34:11.351] [WARN] console - No "skinName" parameter found. Please check out settings.json.template and update your settings.json. Falling back to the default "colibris".
[2021-02-16 19:34:11.352] [INFO] console - Using skin "colibris" in dir: C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\src\static\skins\colibris
[2021-02-16 19:34:11.352] [INFO] console - Session key loaded from: C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\SESSIONKEY.txt
[2021-02-16 19:34:11.352] [WARN] console - DirtyDB is used. This is not recommended for production. File location: C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e\etherpad-lite-develop\var\dirty.db
[2021-02-16 19:34:11.694] [INFO] server - Starting Etherpad...
[2021-02-16 19:34:11.731] [INFO] runNpm - Executing command: npm ls --long --json --depth=0
[2021-02-16 19:34:11.875] [INFO] runNpm - npm --version: 6.14.11
[2021-02-16 19:34:12.178] [INFO] runNpm - Successfully ran command: npm ls --long --json --depth=0
[2021-02-16 19:34:12.178] [INFO] server - Installed plugins:
[2021-02-16 19:34:12.179] [INFO] server - Etherpad is running
```

**To Reproduce**

1. I downloaded changeset d7ed71e from the develop branch.
2. Executed InstallOnWindows.bat (no reported errors).
3. Executed start.bat and the above error was reported.

I tried installing a plugin, ep_copy_paste_images, via the admin plugin manager web page, and it crashed the server.

    [2021-02-17 13:25:09.001] [WARN] console - Can't get git version for server header
    ENOENT: no such file or directory, lstat 'C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop/.git'
    npm WARN saveError ENOENT: no such file or directory, open 'C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\package.json'
    npm notice created a lockfile as package-lock.json. You should commit this file.
    npm WARN enoent ENOENT: no such file or directory, open 'C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\package.json'
    npm WARN etherpad-lite-develop No description
    npm WARN etherpad-lite-develop No repository field.
    npm WARN etherpad-lite-develop No README data
    npm WARN etherpad-lite-develop No license field.
    [2021-02-17 13:25:28.249] [ERROR] console -
    [2021-02-17 13:25:28.256] [INFO] console - + ep_copy_paste_images@0.0.28
    added 1 package from 1 contributor and audited 919 packages in 5.205s
    [2021-02-17 13:25:28.505] [INFO] console -
    73 packages are looking for funding
      run `npm fund` for details
    
    [2021-02-17 13:25:28.506] [INFO] console - found 0 vulnerabilities
    
    [2021-02-17 13:25:28.510] [INFO] runNpm - Executing command: npm ls --long --json --depth=0
    [2021-02-17 13:25:30.562] [ERROR] npm - npm ERR! peer dep missing: ep_etherpad-lite@>=1.8.6, required by ep_copy_paste_images@0.0.28
    [2021-02-17 13:25:30.576] [ERROR] runNpm - npm command failed: npm ls --long --json --depth=0
    [2021-02-17 13:25:30.576] [WARN] console - Error: Command exited non-zero: npm ls --long --json --depth=0
        at exports (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite\node\utils\run_cmd.js:71:25)
        at exports (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite\node\utils\run_npm.js:25:13)
        at Object.exports.getPackages (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite\static\js\pluginfw\plugins.js:73:14)
        at Object.exports.update (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite\static\js\pluginfw\plugins.js:56:34)
        at Object.exports.install (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite\static\js\pluginfw\installer.js:54:19)
    [2021-02-17 13:25:30.577] [INFO] http - Closing HTTP server...

Restarting the server now fails due to dependency issues.

    [2021-02-17 13:47:59.253] [DEBUG] console - Running on Node v12.20.2 (minimum required Node version: 10.17.0)
    [2021-02-17 13:47:59.265] [DEBUG] AbsolutePaths - C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite does not end with "src"
    [2021-02-17 13:47:59.266] [INFO] console - All relative paths will be interpreted relative to the identified Etherpad base dir: C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop
    [2021-02-17 13:47:59.267] [INFO] console - Random string used for versioning assets: cd9d940b
    [2021-02-17 13:47:59.267] [DEBUG] AbsolutePaths - Relative path "settings.json" can be rewritten to "C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\settings.json"
    [2021-02-17 13:47:59.267] [DEBUG] AbsolutePaths - Relative path "credentials.json" can be rewritten to "C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\credentials.json"
    [2021-02-17 13:47:59.273] [INFO] console - settings loaded from: C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\settings.json
    [2021-02-17 13:47:59.275] [INFO] console - No credentials file found in C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\credentials.json. Ignoring.
    [2021-02-17 13:47:59.278] [INFO] console - Using skin "colibris" in dir: C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite\static\skins\colibris
    [2021-02-17 13:47:59.278] [INFO] console - Session key loaded from: C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\SESSIONKEY.txt
    [2021-02-17 13:47:59.815] [INFO] server - Starting Etherpad...
    tedious deprecated The default value for "config.options.validateBulkLoadParameters" will change from `false` to `true` in the next major version of `tedious`. Set the value to `true` or `false` explicitly to silence this message. node_modules\ep_etherpad-lite\node_modules\mssql\lib\tedious\connection-pool.js:62:23
    [2021-02-17 13:47:59.905] [INFO] runNpm - Executing command: npm ls --long --json --depth=0
    [2021-02-17 13:48:00.192] [INFO] runNpm - npm --version: 6.14.11
    [2021-02-17 13:48:01.969] [ERROR] npm - npm ERR! peer dep missing: ep_etherpad-lite@>=1.8.6, required by ep_copy_paste_images@0.0.28
    [2021-02-17 13:48:01.983] [ERROR] runNpm - npm command failed: npm ls --long --json --depth=0
    [2021-02-17 13:48:01.983] [ERROR] server - Error occurred while starting Etherpad
    [2021-02-17 13:48:01.983] [ERROR] server - Error: Command exited non-zero: npm ls --long --json --depth=0
        at exports (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite\node\utils\run_cmd.js:71:25)
        at exports (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite\node\utils\run_npm.js:25:13)
        at Object.exports.getPackages (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite\static\js\pluginfw\plugins.js:73:14)
        at Object.exports.update (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite\static\js\pluginfw\plugins.js:56:34)
        at Object.exports.start (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite\node\server.js:137:19)
        at processTicksAndRejections (internal/process/task_queues.js:97:5)
    [2021-02-17 13:48:01.984] [INFO] server - Exiting...
    [2021-02-17 13:48:01.984] [INFO] server - Waiting for Node.js to exit...
    [2021-02-17 13:48:06.989] [ERROR] server - Something that should have been cleaned up during the shutdown hook (such as a timer, worker thread, or open connection) is preventing Node.js from exiting
    [2021-02-17 13:48:06.990] [INFO] console - [WTF Node?] open handles:
    [2021-02-17 13:48:06.992] [INFO] console - - File descriptors: (note: stdio always exists)
    [2021-02-17 13:48:06.992] [INFO] console -   - fd 1 (tty) (stdio)
    [2021-02-17 13:48:06.992] [INFO] console -   - fd 2 (tty) (stdio)
    [2021-02-17 13:48:06.992] [INFO] console - - Sockets:
    [2021-02-17 13:48:06.993] [INFO] console -   - 127.0.0.1:59718 -> 127.0.0.1:1433
    [2021-02-17 13:48:06.993] [INFO] console - - Intervals:
    [2021-02-17 13:48:06.994] [INFO] console -   - (1000 ~ 1000 ms) (anonymous) @ C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite\node_modules\tarn\dist\Pool.js:451
    [2021-02-17 13:48:06.994] [ERROR] server - Forcing an unclean exit...

I assume this is probably related to moving the `src` folder. After installing the addon, I now have a `package-log.json1` file in the site root (I can't remember if this was here before). `node_modules` has a folder for `ep_copy_paste_images`. I tried moving this folder into `node_modules/ep_etherpad-lite/node_moduels', but that causes this crash.

    [2021-02-17 13:57:25.568] [DEBUG] console - Running on Node v12.20.2 (minimum required Node version: 10.17.0)
    [2021-02-17 13:57:25.583] [DEBUG] AbsolutePaths - C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite does not end with "src"
    [2021-02-17 13:57:25.583] [INFO] console - All relative paths will be interpreted relative to the identified Etherpad base dir: C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop
    [2021-02-17 13:57:25.584] [INFO] console - Random string used for versioning assets: f95d1d38
    [2021-02-17 13:57:25.585] [DEBUG] AbsolutePaths - Relative path "settings.json" can be rewritten to "C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\settings.json"
    [2021-02-17 13:57:25.585] [DEBUG] AbsolutePaths - Relative path "credentials.json" can be rewritten to "C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\credentials.json"
    [2021-02-17 13:57:25.591] [INFO] console - settings loaded from: C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\settings.json
    [2021-02-17 13:57:25.592] [INFO] console - No credentials file found in C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\credentials.json. Ignoring.
    [2021-02-17 13:57:25.594] [INFO] console - Using skin "colibris" in dir: C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite\static\skins\colibris
    [2021-02-17 13:57:25.594] [INFO] console - Session key loaded from: C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\SESSIONKEY.txt
    [2021-02-17 13:57:26.111] [INFO] server - Starting Etherpad...
    tedious deprecated The default value for "config.options.validateBulkLoadParameters" will change from `false` to `true` in the next major version of `tedious`. Set the value to `true` or `false` explicitly to silence this message. node_modules\ep_etherpad-lite\node_modules\mssql\lib\tedious\connection-pool.js:62:23
    [2021-02-17 13:57:26.197] [INFO] runNpm - Executing command: npm ls --long --json --depth=0
    [2021-02-17 13:57:26.482] [INFO] runNpm - npm --version: 6.14.11
    [2021-02-17 13:57:27.928] [ERROR] npm - npm ERR! missing: ep_copy_paste_images@0.0.28, required by etherpad-lite-develop
    [2021-02-17 13:57:27.942] [ERROR] runNpm - npm command failed: npm ls --long --json --depth=0
    [2021-02-17 13:57:27.943] [ERROR] server - Error occurred while starting Etherpad
    [2021-02-17 13:57:27.943] [ERROR] server - Error: Command exited non-zero: npm ls --long --json --depth=0
        at exports (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite\node\utils\run_cmd.js:71:25)
        at exports (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite\node\utils\run_npm.js:25:13)
        at Object.exports.getPackages (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite\static\js\pluginfw\plugins.js:73:14)
        at Object.exports.update (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite\static\js\pluginfw\plugins.js:56:34)
        at Object.exports.start (C:\Users\alexe\Downloads\etherpad-lite-develop-d7ed71e-2\etherpad-lite-develop\node_modules\ep_etherpad-lite\node\server.js:137:19)
        at processTicksAndRejections (internal/process/task_queues.js:97:5)
    [2021-02-17 13:57:27.943] [INFO] server - Exiting...

`node_modules\ep_etherpad-lite\package.json` was not modified by installing this plugin.

I'm just working out of a folder right now, no change control, so I can't see the entire delta of this action. Any ideas? I have to run to a meeting, but I can debug later this afternoon if the answer isn't obvious.

I'm making progress with Azure. It's now starting a workflow in Github to do a deployment. I'm running into an issue with the package.json file. I think it's looking for this file in the site root, but it's located in `src`.

Is it normal to have the packages.json in a non-root folder? Is this related to the bug fix you're working on?

Here's the relevant output of the workflow.

    Run npm install
      npm install
      npm run build --if-present
      npm run test --if-present
      shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
    npm WARN saveError ENOENT: no such file or directory, open 'D:\a\accountname\webappname\package.json'
    npm notice created a lockfile as package-lock.json. You should commit this file.
    npm WARN enoent ENOENT: no such file or directory, open 'D:\a\accountname\webappname\package.json'
    
    npm ERR! code ENOENT
    npm ERR! syscall open
    npm ERR! path D:\a\accountname\webappname\package.json
    npm ERR! errno -4058
    npm ERR! enoent ENOENT: no such file or directory, open 'D:\a\westfaliaetherpad\westfaliaetherpad\package.json'
    npm ERR! enoent This is related to npm not being able to find a file.
    npm ERR! enoent

I'm making progress with Azure. It's now starting a workflow in Github to do a deployment. I'm running into an issue with the package.json file. I think it's looking for this file in the site root, but it's located in `src`.

Is it normal to have the packages.json in a non-root folder? Is this related to the bug fix you're working on?

Here's the relevant output of the workflow.

    Run npm install
      npm install
      npm run build --if-present
      npm run test --if-present
      shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
    npm WARN saveError ENOENT: no such file or directory, open 'D:\a\accountname\webappname\package.json'
    npm notice created a lockfile as package-lock.json. You should commit this file.
    npm WARN enoent ENOENT: no such file or directory, open 'D:\a\accountname\webappname\package.json'
    
    npm ERR! code ENOENT
    npm ERR! syscall open
    npm ERR! path D:\a\accountname\webappname\package.json
    npm ERR! errno -4058
    npm ERR! enoent ENOENT: no such file or directory, open 'D:\a\accountname\webappname\package.json'
    npm ERR! enoent This is related to npm not being able to find a file.
    npm ERR! enoent

Awesome! Up and running locally with a new folder using these commands:

    mkdir etherpad-dev
    cd etherpad-dev\etherpad-lite
    git clone https://github.com/ether/etherpad-lite.git
    cd src && npm i && cd ..
    node src\node\server.js

At this point, I could load the site on localhost:9001.

Then I created a new repository in my personal Github, set the remote url, and checked in the latest.

    git remote -v -- Confirm that it points to the official Etherpad-Lite repo
    git remote set-url origin https://github.com/username/repo.git -- Update to point to my repo
    git remote -v -- Confirm that it points to my repo
    git add --all
    git commit -m "initial"
    git push

I have an Azure App Service set up with a Windows server. I'm using the Deployment Center to create a Github integration for CI/CD. It creates the connection to the Github repo fine and creates a `.yml` file in `.github\workflows`. The contents are below. This workflow is failing on the Github side.

    # Docs for the Azure Web Apps Deploy action: https://github.com/Azure/webapps-deploy
    # More GitHub Actions for Azure: https://github.com/Azure/actions
    
    name: Build and deploy Node.js app to Azure Web App - mysite-etherpad
    
    on:
      push:
        branches:
          - develop
      workflow_dispatch:
    
    jobs:
      build-and-deploy:
        runs-on: windows-latest
    
        steps:
        - uses: actions/checkout@master
    
        - name: Set up Node.js version
          uses: actions/setup-node@v1
          with:
            node-version: '12.13.0'
    
        - name: npm install, build, and test
          run: |
            npm install
            npm run build --if-present
            npm run test --if-present
    
        - name: 'Deploy to Azure Web App'
          uses: azure/webapps-deploy@v2
          with:
            app-name: 'mysite-etherpad'
            slot-name: 'production'
            publish-profile: ${{ secrets.AzureAppService_PublishProfile_4fc8dd232ce24a5d9179defc7ba3fca6 }}
            package: .

This was failing because it couldn't find package.json. I modified three lines under `name: npm install, build and test`:

    npm --prefix src install
    npm --prefix src run build --if-present
    # npm --prefix src run test --if-present

This still failed in Github workflow, but it wasn't a fatal error and the code was deployed to Azure.

Once in Azure, I had some web.config issues that I think I resolved by deleting the web.config. There were issues caused by duplicated `iisnode` nodes.

Now it appears to start running the server, but it stops without an error code and I'm returned an HTTP 500.1001 error: iisnode was unable to establish named pipe connection to the node.exe process. Below are the best logs I can find on the server. Any suggestions? I'm trying to get support from Azure again.

    [36m[2021-02-17 20:38:22.572] [DEBUG] console - [39mRunning on Node v12.13.0 (minimum required Node version: 10.17.0)
    [32m[2021-02-17 20:38:22.759] [INFO] console - [39mAll relative paths will be interpreted relative to the identified Etherpad base dir: C:\home\site\wwwroot
    [32m[2021-02-17 20:38:22.759] [INFO] console - [39mRandom string used for versioning assets: b1e64664
    [36m[2021-02-17 20:38:22.775] [DEBUG] AbsolutePaths - [39mRelative path "settings.json" can be rewritten to "C:\home\site\wwwroot\settings.json"
    [36m[2021-02-17 20:38:22.791] [DEBUG] AbsolutePaths - [39mRelative path "credentials.json" can be rewritten to "C:\home\site\wwwroot\credentials.json"
    [32m[2021-02-17 20:38:22.827] [INFO] console - [39msettings loaded from: C:\home\site\wwwroot\settings.json
    [32m[2021-02-17 20:38:22.838] [INFO] console - [39mNo credentials file found in C:\home\site\wwwroot\credentials.json. Ignoring.
    [32m[2021-02-17 20:38:22.838] [INFO] console - [39mUsing skin "colibris" in dir: C:\home\site\wwwroot\src\static\skins\colibris
    [32m[2021-02-17 20:38:22.853] [INFO] console - [39mSession key loaded from: C:\home\site\wwwroot\SESSIONKEY.txt
    [31m[2021-02-17 20:38:37.541] [ERROR] console - [39m(node:14704) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
    [32m[2021-02-17 20:38:39.855] [INFO] runNpm - [39mnpm --version: 6.14.11

Progress! Azure does not support symlinks. I copied `src/bin` to `bin`. 

There is a development console available for the App Service via the Azure Portal. I used this console to manually run the server and I got the actual error logs:

    [36m[2021-02-19 16:15:06.540] [DEBUG] console - [39mRunning on Node v12.13.0 (minimum required Node version: 10.17.0)
    [32m[2021-02-19 16:15:06.697] [INFO] console - [39mAll relative paths will be interpreted relative to the identified Etherpad base dir: C:\home\site\wwwroot
    [32m[2021-02-19 16:15:06.712] [INFO] console - [39mRandom string used for versioning assets: 43551526
    [36m[2021-02-19 16:15:06.712] [DEBUG] AbsolutePaths - [39mRelative path "settings.json" can be rewritten to "C:\home\site\wwwroot\settings.json"
    [36m[2021-02-19 16:15:06.712] [DEBUG] AbsolutePaths - [39mRelative path "credentials.json" can be rewritten to "C:\home\site\wwwroot\credentials.json"
    [32m[2021-02-19 16:15:06.743] [INFO] console - [39msettings loaded from: C:\home\site\wwwroot\settings.json
    [32m[2021-02-19 16:15:06.743] [INFO] console - [39mNo credentials file found in C:\home\site\wwwroot\credentials.json. Ignoring.
    [32m[2021-02-19 16:15:06.762] [INFO] console - [39mUsing skin "colibris" in dir: C:\home\site\wwwroot\src\static\skins\colibris
    [32m[2021-02-19 16:15:06.775] [INFO] console - [39mSession key loaded from: C:\home\site\wwwroot\SESSIONKEY.txt
    [33m[2021-02-19 16:15:06.775] [WARN] console - [39mDirtyDB is used. This is not recommended for production. File location: C:\home\site\wwwroot\var\dirty.db
    [33m[2021-02-19 16:15:06.775] [WARN] console - [39mThe settings file contains an empty string ("") for the "ip" parameter. The "port" parameter will be interpreted as the path to a Unix socket to bind at.
    [32m[2021-02-19 16:15:13.275] [INFO] server - [39mStarting Etherpad...
    [32m[2021-02-19 16:15:13.666] [INFO] plugins - [39mRunning npm to get a list of installed plugins...
    [32m[2021-02-19 16:15:18.651] [INFO] plugins - [39mnpm --version: 6.14.11
    [31m[2021-02-19 16:15:51.792] [ERROR] runCmd|npm - [39mnpm ERR! missing: ep_etherpad-lite@file:src, required by wwwroot
    [31m[2021-02-19 16:15:51.792] [ERROR] runCmd|npm - [39mnpm ERR! missing: eslint@7.18.0, required by ep_etherpad-lite@file:src
    [... Error logs trimmed by Alex ...]
    [31m[2021-02-19 16:15:51.932] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: underscore@1.12.0 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\underscore
    [31m[2021-02-19 16:15:51.932] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: unorm@1.4.1 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\unorm
    [31m[2021-02-19 16:15:51.932] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: wtfnode@0.8.4 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\wtfnode
    [31m[2021-02-19 16:15:51.948] [ERROR] server - [39mError occurred while starting Etherpad
    [31m[2021-02-19 16:15:51.948] [ERROR] server - [39mError: Command exited with code 1: npm ls --long --json --depth=0 --no-production
        at exports (C:\home\site\wwwroot\src\node\utils\run_cmd.js:119:25)
        at Object.exports.getPackages (C:\home\site\wwwroot\src\static\js\pluginfw\plugins.js:93:48)
        at Object.exports.update (C:\home\site\wwwroot\src\static\js\pluginfw\plugins.js:67:34)
        at Object.exports.start (C:\home\site\wwwroot\src\node\server.js:140:19)
        at processTicksAndRejections (internal/process/task_queues.js:93:5)
    [32m[2021-02-19 16:15:51.948] [INFO] server - [39mExiting...
    [32m[2021-02-19 16:15:51.948] [INFO] server - [39mWaiting for Node.js to exit...

I tried to install the dependencies with `npm i`

    cd src
    npm i

Output:

    npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.3 (node_modules\fsevents):
    npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"ia32"})

I ran `server.js` again. Same log entries: all modules are missing and then all modules are extraneous.

Hey @RyanHill-MSFT. Thanks for the info. The Node.JS best practices link is new to me, and that looks very useful. At this point I'm running into application issues that I think are related to the symlinks. I think setting `src` as the working folder won't work because NPM is used to manage plugins for this project as well.

@JohnMcLear, I mentioned in my last post that I has copied `src/bin` to `bin`, but I forgot about the symlink at `node_moduels/ep_etherpad-lite`. Since my last poste, I performed the following actions and I'm getting the same issue: all of the modules required by ep_etherpad-lite are both missing and extraneous. I'll paste the full log below.

    Deleted `node_modules\ep_etherpad-lite`
    Created a new folder at `node_modules\ep_etherpad-lite`
    Copied the contents (recursively) of `src` into `node_modules\ep_etherpad-lite`
    node src\node\server.js

Output of starting the server:

    [36m[2021-02-19 18:01:06.186] [DEBUG] console - [39mRunning on Node v12.13.0 (minimum required Node version: 10.17.0)
    [32m[2021-02-19 18:01:06.842] [INFO] console - [39mAll relative paths will be interpreted relative to the identified Etherpad base dir: C:\home\site\wwwroot
    [32m[2021-02-19 18:01:06.842] [INFO] console - [39mRandom string used for versioning assets: aadf2fba
    [36m[2021-02-19 18:01:06.858] [DEBUG] AbsolutePaths - [39mRelative path "settings.json" can be rewritten to "C:\home\site\wwwroot\settings.json"
    [36m[2021-02-19 18:01:06.858] [DEBUG] AbsolutePaths - [39mRelative path "credentials.json" can be rewritten to "C:\home\site\wwwroot\credentials.json"
    [32m[2021-02-19 18:01:06.889] [INFO] console - [39msettings loaded from: C:\home\site\wwwroot\settings.json
    [32m[2021-02-19 18:01:06.889] [INFO] console - [39mNo credentials file found in C:\home\site\wwwroot\credentials.json. Ignoring.
    [32m[2021-02-19 18:01:06.905] [INFO] console - [39mUsing skin "colibris" in dir: C:\home\site\wwwroot\src\static\skins\colibris
    [32m[2021-02-19 18:01:06.924] [INFO] console - [39mSession key loaded from: C:\home\site\wwwroot\SESSIONKEY.txt
    [33m[2021-02-19 18:01:06.924] [WARN] console - [39mDirtyDB is used. This is not recommended for production. File location: C:\home\site\wwwroot\var\dirty.db
    [33m[2021-02-19 18:01:06.924] [WARN] console - [39mThe settings file contains an empty string ("") for the "ip" parameter. The "port" parameter will be interpreted as the path to a Unix socket to bind at.
    [32m[2021-02-19 18:01:16.826] [INFO] server - [39mStarting Etherpad...
    [32m[2021-02-19 18:01:16.967] [INFO] plugins - [39mRunning npm to get a list of installed plugins...
    [32m[2021-02-19 18:01:26.144] [INFO] plugins - [39mnpm --version: 6.14.11
    [31m[2021-02-19 18:01:51.439] [ERROR] runCmd|npm - [39mnpm ERR! missing: ep_etherpad-lite@file:src, required by wwwroot
    [31m[2021-02-19 18:01:51.439] [ERROR] runCmd|npm - [39mnpm ERR! missing: eslint@7.18.0, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.439] [ERROR] runCmd|npm - [39mnpm ERR! missing: eslint-config-etherpad@1.0.24, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.439] [ERROR] runCmd|npm - [39mnpm ERR! missing: eslint-plugin-eslint-comments@3.2.0, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.439] [ERROR] runCmd|npm - [39mnpm ERR! missing: eslint-plugin-mocha@8.0.0, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: eslint-plugin-node@11.1.0, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: eslint-plugin-prefer-arrow@1.2.3, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: eslint-plugin-promise@4.2.1, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: eslint-plugin-you-dont-need-lodash-underscore@6.10.0, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: etherpad-cli-client@0.0.9, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: mocha@7.1.2, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: mocha-froth@0.2.10, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: set-cookie-parser@2.4.7, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: sinon@9.2.4, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: supertest@4.0.2, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: wd@1.12.1, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: async@3.2.0, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: async-stacktrace@0.0.2, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: channels@0.0.4, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: cheerio@0.22.0, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: clean-css@4.2.3, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: cookie-parser@1.4.5, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: cross-spawn@7.0.3, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: ejs@3.1.6, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: etherpad-require-kernel@1.0.9, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.453] [ERROR] runCmd|npm - [39mnpm ERR! missing: etherpad-yajsml@0.0.2, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.472] [ERROR] runCmd|npm - [39mnpm ERR! missing: express@4.17.1, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.472] [ERROR] runCmd|npm - [39mnpm ERR! missing: express-rate-limit@5.2.3, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.472] [ERROR] runCmd|npm - [39mnpm ERR! missing: express-session@1.17.1, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.472] [ERROR] runCmd|npm - [39mnpm ERR! missing: find-root@1.1.0, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.472] [ERROR] runCmd|npm - [39mnpm ERR! missing: formidable@1.2.1, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.472] [ERROR] runCmd|npm - [39mnpm ERR! missing: http-errors@1.8.0, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.472] [ERROR] runCmd|npm - [39mnpm ERR! missing: js-cookie@2.2.1, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.472] [ERROR] runCmd|npm - [39mnpm ERR! missing: jsonminify@0.4.1, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.472] [ERROR] runCmd|npm - [39mnpm ERR! missing: languages4translatewiki@0.1.3, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.472] [ERROR] runCmd|npm - [39mnpm ERR! missing: lodash.clonedeep@4.5.0, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.472] [ERROR] runCmd|npm - [39mnpm ERR! missing: log4js@0.6.35, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.472] [ERROR] runCmd|npm - [39mnpm ERR! missing: measured-core@1.51.1, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.472] [ERROR] runCmd|npm - [39mnpm ERR! missing: mime-types@2.1.28, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.472] [ERROR] runCmd|npm - [39mnpm ERR! missing: nodeify@1.0.1, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.472] [ERROR] runCmd|npm - [39mnpm ERR! missing: npm@6.14.11, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.472] [ERROR] runCmd|npm - [39mnpm ERR! missing: openapi-backend@2.4.1, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.472] [ERROR] runCmd|npm - [39mnpm ERR! missing: proxy-addr@2.0.6, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.485] [ERROR] runCmd|npm - [39mnpm ERR! missing: rate-limiter-flexible@2.2.1, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.485] [ERROR] runCmd|npm - [39mnpm ERR! missing: rehype@10.0.0, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.485] [ERROR] runCmd|npm - [39mnpm ERR! missing: rehype-minify-whitespace@4.0.5, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.485] [ERROR] runCmd|npm - [39mnpm ERR! missing: request@2.88.2, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.485] [ERROR] runCmd|npm - [39mnpm ERR! missing: resolve@1.19.0, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.485] [ERROR] runCmd|npm - [39mnpm ERR! missing: security@1.0.0, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.485] [ERROR] runCmd|npm - [39mnpm ERR! missing: semver@5.7.1, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.485] [ERROR] runCmd|npm - [39mnpm ERR! missing: socket.io@2.4.1, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.485] [ERROR] runCmd|npm - [39mnpm ERR! missing: terser@4.8.0, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.485] [ERROR] runCmd|npm - [39mnpm ERR! missing: threads@1.6.3, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.485] [ERROR] runCmd|npm - [39mnpm ERR! missing: tiny-worker@2.3.0, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.485] [ERROR] runCmd|npm - [39mnpm ERR! missing: tinycon@0.0.1, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.485] [ERROR] runCmd|npm - [39mnpm ERR! missing: ueberdb2@1.2.5, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.485] [ERROR] runCmd|npm - [39mnpm ERR! missing: underscore@1.12.0, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.485] [ERROR] runCmd|npm - [39mnpm ERR! missing: unorm@1.4.1, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.485] [ERROR] runCmd|npm - [39mnpm ERR! missing: wtfnode@0.8.4, required by ep_etherpad-lite@file:src
    [31m[2021-02-19 18:01:51.485] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: eslint@7.18.0 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\eslint
    [31m[2021-02-19 18:01:51.485] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: eslint-config-etherpad@1.0.24 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\eslint-config-etherpad
    [31m[2021-02-19 18:01:51.503] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: eslint-plugin-eslint-comments@3.2.0 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\eslint-plugin-eslint-comments
    [31m[2021-02-19 18:01:51.503] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: eslint-plugin-mocha@8.0.0 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\eslint-plugin-mocha
    [31m[2021-02-19 18:01:51.503] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: eslint-plugin-node@11.1.0 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\eslint-plugin-node
    [31m[2021-02-19 18:01:51.503] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: eslint-plugin-prefer-arrow@1.2.3 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\eslint-plugin-prefer-arrow
    [31m[2021-02-19 18:01:51.503] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: eslint-plugin-promise@4.2.1 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\eslint-plugin-promise
    [31m[2021-02-19 18:01:51.503] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: eslint-plugin-you-dont-need-lodash-underscore@6.10.0 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\eslint-plugin-you-dont-need-lodash-underscore
    [31m[2021-02-19 18:01:51.503] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: etherpad-cli-client@0.0.9 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\etherpad-cli-client
    [31m[2021-02-19 18:01:51.503] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: mocha@7.1.2 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\mocha
    [31m[2021-02-19 18:01:51.503] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: mocha-froth@0.2.10 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\mocha-froth
    [31m[2021-02-19 18:01:51.503] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: set-cookie-parser@2.4.7 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\set-cookie-parser
    [31m[2021-02-19 18:01:51.503] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: sinon@9.2.4 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\sinon
    [31m[2021-02-19 18:01:51.503] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: supertest@4.0.2 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\supertest
    [31m[2021-02-19 18:01:51.503] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: wd@1.12.1 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\wd
    [31m[2021-02-19 18:01:51.503] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: async@3.2.0 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\async
    [31m[2021-02-19 18:01:51.503] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: async-stacktrace@0.0.2 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\async-stacktrace
    [31m[2021-02-19 18:01:51.503] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: channels@0.0.4 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\channels
    [31m[2021-02-19 18:01:51.503] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: cheerio@0.22.0 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\cheerio
    [31m[2021-02-19 18:01:51.516] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: clean-css@4.2.3 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\clean-css
    [31m[2021-02-19 18:01:51.516] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: cookie-parser@1.4.5 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\cookie-parser
    [31m[2021-02-19 18:01:51.516] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: cross-spawn@7.0.3 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\cross-spawn
    [31m[2021-02-19 18:01:51.516] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: ejs@3.1.6 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\ejs
    [31m[2021-02-19 18:01:51.516] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: etherpad-require-kernel@1.0.9 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\etherpad-require-kernel
    [31m[2021-02-19 18:01:51.516] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: etherpad-yajsml@0.0.2 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\etherpad-yajsml
    [31m[2021-02-19 18:01:51.516] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: express@4.17.1 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\express
    [31m[2021-02-19 18:01:51.516] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: express-rate-limit@5.2.3 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\express-rate-limit
    [31m[2021-02-19 18:01:51.516] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: express-session@1.17.1 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\express-session
    [31m[2021-02-19 18:01:51.516] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: find-root@1.1.0 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\find-root
    [31m[2021-02-19 18:01:51.516] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: formidable@1.2.1 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\formidable
    [31m[2021-02-19 18:01:51.516] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: http-errors@1.8.0 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\http-errors
    [31m[2021-02-19 18:01:51.516] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: js-cookie@2.2.1 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\js-cookie
    [31m[2021-02-19 18:01:51.516] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: jsonminify@0.4.1 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\jsonminify
    [31m[2021-02-19 18:01:51.516] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: languages4translatewiki@0.1.3 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\languages4translatewiki
    [31m[2021-02-19 18:01:51.532] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: lodash.clonedeep@4.5.0 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\lodash.clonedeep
    [31m[2021-02-19 18:01:51.532] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: log4js@0.6.35 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\log4js
    [31m[2021-02-19 18:01:51.532] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: measured-core@1.51.1 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\measured-core
    [31m[2021-02-19 18:01:51.532] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: mime-types@2.1.28 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\mime-types
    [31m[2021-02-19 18:01:51.532] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: nodeify@1.0.1 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\nodeify
    [31m[2021-02-19 18:01:51.532] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: npm@6.14.11 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\npm
    [31m[2021-02-19 18:01:51.532] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: openapi-backend@2.4.1 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\openapi-backend
    [31m[2021-02-19 18:01:51.532] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: proxy-addr@2.0.6 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\proxy-addr
    [31m[2021-02-19 18:01:51.532] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: rate-limiter-flexible@2.2.1 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\rate-limiter-flexible
    [31m[2021-02-19 18:01:51.532] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: rehype@10.0.0 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\rehype
    [31m[2021-02-19 18:01:51.532] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: rehype-minify-whitespace@4.0.5 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\rehype-minify-whitespace
    [31m[2021-02-19 18:01:51.532] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: request@2.88.2 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\request
    [31m[2021-02-19 18:01:51.532] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: resolve@1.19.0 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\resolve
    [31m[2021-02-19 18:01:51.532] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: security@1.0.0 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\security
    [31m[2021-02-19 18:01:51.532] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: semver@5.7.1 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\semver
    [31m[2021-02-19 18:01:51.532] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: socket.io@2.4.1 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\socket.io
    [31m[2021-02-19 18:01:51.547] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: terser@4.8.0 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\terser
    [31m[2021-02-19 18:01:51.547] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: threads@1.6.3 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\threads
    [31m[2021-02-19 18:01:51.547] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: tiny-worker@2.3.0 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\tiny-worker
    [31m[2021-02-19 18:01:51.547] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: tinycon@0.0.1 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\tinycon
    [31m[2021-02-19 18:01:51.547] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: ueberdb2@1.2.5 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\ueberdb2
    [31m[2021-02-19 18:01:51.547] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: underscore@1.12.0 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\underscore
    [31m[2021-02-19 18:01:51.547] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: unorm@1.4.1 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\unorm
    [31m[2021-02-19 18:01:51.547] [ERROR] runCmd|npm - [39mnpm ERR! extraneous: wtfnode@0.8.4 C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node_modules\wtfnode
    [31m[2021-02-19 18:01:51.578] [ERROR] server - [39mError occurred while starting Etherpad
    [31m[2021-02-19 18:01:51.578] [ERROR] server - [39mError: Command exited with code 1: npm ls --long --json --depth=0 --no-production
        at exports (C:\home\site\wwwroot\src\node\utils\run_cmd.js:119:25)
        at Object.exports.getPackages (C:\home\site\wwwroot\src\static\js\pluginfw\plugins.js:93:48)
        at Object.exports.update (C:\home\site\wwwroot\src\static\js\pluginfw\plugins.js:67:34)
        at Object.exports.start (C:\home\site\wwwroot\src\node\server.js:140:19)
        at processTicksAndRejections (internal/process/task_queues.js:93:5)
    [32m[2021-02-19 18:01:51.578] [INFO] server - [39mExiting...
    [32m[2021-02-19 18:01:51.578] [INFO] server - [39mWaiting for Node.js to exit...

> Running the sever with node via the console shows that it's trying to bind to 9001.

What is the value of the `PORT` environment variable? I'm not familiar with Windows shells, but on POSIX systems you can run `echo $PORT` to check.

> ```
> runCmd|npm - [39mnpm ERR! missing: ep_etherpad-lite@file:src, required by wwwroot
> ```

This indicates that you somehow ended up with a `package.json` and/or `package-lock.json` in the top level of your Etherpad directory. What version of npm are you using (`npm --version`)?

Thanks @RyanHill-MSFT! Quick question: the application logs which version of NPM is being used. It logs `version 1.1.37` (you can see this in the logs below). When I check the version via the Console, it reports `version 6.12.0`. Do you have any insight here? Where would NPM 1.1.37 be coming from? I've recorded my actions, the logs, and the web.config below.

I redeployed the site. The application ships a `web.config` in a subfolder: I deleted `node_modules/ep_etherpad-lite/web.config`.

Here's the `web.config` in the site root:

    <configuration>
      <system.webServer>
    
        <handlers>
          <add name="iisnode" path="node_modules/ep_etherpad-lite/node/server.js" verb="*" modules="iisnode" />
        </handlers>
    
        <rewrite>
            <rules>
                <rule name="LogFile" patternSyntax="ECMAScript" stopProcessing="true">
                    <match url="iisnode"/>
                    <action type="Rewrite" url="node_modules/ep_etherpad-lite/node/iisnode" />
                </rule>
                <rule name="NodeInspector" patternSyntax="ECMAScript" stopProcessing="true">                    
                    <match url="^server.js\/debug[\/]?" />
                </rule>
                
                <rule name="StaticContent">
                     <action type="Rewrite" url="public{{REQUEST_URI}}"/>
                </rule>
                <rule name="DynamicContent">
                     <conditions>
                          <add input="{{REQUEST_FILENAME}}" matchType="IsFile" negate="True"/>
                     </conditions>
                    <action type="Rewrite" url="node_modules/ep_etherpad-lite/node/server.js" />
                </rule>
            </rules>
        </rewrite>
    
        <iisnode node_env="%node_env%"
            nodeProcessCountPerApplication="1"
            maxConcurrentRequestsPerProcess="1024"
            maxNamedPipeConnectionRetry="100"
            namedPipeConnectionRetryDelay="250"
            maxNamedPipeConnectionPoolSize="512"
            maxNamedPipePooledConnectionAge="30000"
            asyncCompletionThreadCount="0"
            initialRequestBufferSize="4096"
            maxRequestBufferSize="65536"
            uncFileChangesPollingInterval="5000"
            gracefulShutdownTimeout="60000"
            loggingEnabled="true"
            logDirectory="iisnode"
            debuggingEnabled="true"
            debugHeaderEnabled="false"
            debuggerPortRange="5058-6058"
            debuggerPathSegment="debug"
            maxLogFileSizeInKB="128"
            maxTotalLogFileSizeInKB="1024"
            maxLogFiles="20"
            devErrorsEnabled="true"
            flushResponse="false"
            enableXFF="false"
            promoteServerVars=""
            configOverrides="iisnode.yml"
            watchedFiles="web.config;*.js"
            nodeProcessCommandLine="C:\home\site\wwwroot\node.exe"
            />

      </system.webServer>
    </configuration>

I didn't understand the purpose of `<iisnode>` before, so this has definitely helped! I used the above configuration and got these results:

    [36m[2021-02-22 16:46:13.395] [DEBUG] console - [39mRunning on Node v12.20.2 (minimum required Node version: 10.17.0)
    [36m[2021-02-22 16:46:13.613] [DEBUG] AbsolutePaths - [39mC:\home\site\wwwroot\node_modules\ep_etherpad-lite does not end with "src"
    [32m[2021-02-22 16:46:13.613] [INFO] console - [39mAll relative paths will be interpreted relative to the identified Etherpad base dir: C:\home\site\wwwroot
    [32m[2021-02-22 16:46:13.629] [INFO] console - [39mRandom string used for versioning assets: c3a38752
    [36m[2021-02-22 16:46:13.645] [DEBUG] AbsolutePaths - [39mRelative path "settings.json" can be rewritten to "C:\home\site\wwwroot\settings.json"
    [36m[2021-02-22 16:46:13.645] [DEBUG] AbsolutePaths - [39mRelative path "credentials.json" can be rewritten to "C:\home\site\wwwroot\credentials.json"
    [32m[2021-02-22 16:46:13.676] [INFO] console - [39msettings loaded from: C:\home\site\wwwroot\settings.json
    [32m[2021-02-22 16:46:13.692] [INFO] console - [39mNo credentials file found in C:\home\site\wwwroot\credentials.json. Ignoring.
    [32m[2021-02-22 16:46:13.707] [INFO] console - [39mUsing skin "colibris" in dir: C:\home\site\wwwroot\src\static\skins\colibris
    [32m[2021-02-22 16:46:13.723] [INFO] console - [39mSession key loaded from: C:\home\site\wwwroot\SESSIONKEY.txt
    [33m[2021-02-22 16:46:13.738] [WARN] console - [39mDirtyDB is used. This is not recommended for production. File location: C:\home\site\wwwroot\var\dirty.db
    [31m[2021-02-22 16:46:23.754] [ERROR] console - [39m(node:34760) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
    [32m[2021-02-22 16:46:25.004] [INFO] runNpm - [39mnpm --version: 1.1.37

In the browser, this was returned:

    iisnode encountered an error when processing the request.
    
    HRESULT: 0x2
    HTTP status: 500
    HTTP subStatus: 1001
    HTTP reason: Internal Server Error
    
    You are receiving this HTTP 200 response because system.webServer/iisnode/@devErrorsEnabled configuration setting is 'true'.
    
    In addition to the log of stdout and stderr of the node.exe process, consider using debugging and ETW traces to further diagnose the problem.
    
    The node.exe process has not written any information to stderr or iisnode was unable to capture this information. Frequent reason is that the iisnode module is unable to create a log file to capture stdout and stderr output from node.exe. Please check that the identity of the IIS application pool running the node.js application has read and write access permissions to the directory on the server where the node.js application is located. Alternatively you can disable logging by setting system.webServer/iisnode/@loggingEnabled element of web.config to 'false'.

I increased the `maxNamedPipeConnectionRetry` to 1000 and refreshed the browser. The browser returned a 500 error this time.

    500 - The request timed out. The web server failed to respond within the specified time.

Server side, I have the same application logs that stop at the same entry.

    [36m[2021-02-22 16:37:39.663] [DEBUG] console - [39mRunning on Node v12.20.2 (minimum required Node version: 10.17.0)
    [36m[2021-02-22 16:37:39.851] [DEBUG] AbsolutePaths - [39mC:\home\site\wwwroot\node_modules\ep_etherpad-lite does not end with "src"
    [32m[2021-02-22 16:37:39.866] [INFO] console - [39mAll relative paths will be interpreted relative to the identified Etherpad base dir: C:\home\site\wwwroot
    [32m[2021-02-22 16:37:39.882] [INFO] console - [39mRandom string used for versioning assets: 0adb147c
    [36m[2021-02-22 16:37:39.898] [DEBUG] AbsolutePaths - [39mRelative path "settings.json" can be rewritten to "C:\home\site\wwwroot\settings.json"
    [36m[2021-02-22 16:37:39.913] [DEBUG] AbsolutePaths - [39mRelative path "credentials.json" can be rewritten to "C:\home\site\wwwroot\credentials.json"
    [32m[2021-02-22 16:37:39.960] [INFO] console - [39msettings loaded from: C:\home\site\wwwroot\settings.json
    [32m[2021-02-22 16:37:39.976] [INFO] console - [39mNo credentials file found in C:\home\site\wwwroot\credentials.json. Ignoring.
    [32m[2021-02-22 16:37:39.992] [INFO] console - [39mUsing skin "colibris" in dir: C:\home\site\wwwroot\src\static\skins\colibris
    [32m[2021-02-22 16:37:40.007] [INFO] console - [39mSession key loaded from: C:\home\site\wwwroot\SESSIONKEY.txt
    [33m[2021-02-22 16:37:40.023] [WARN] console - [39mDirtyDB is used. This is not recommended for production. File location: C:\home\site\wwwroot\var\dirty.db
    [31m[2021-02-22 16:37:50.304] [ERROR] console - [39m(node:37536) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
    [32m[2021-02-22 16:37:51.570] [INFO] runNpm - [39mnpm --version: 1.1.37

@JohnMcLear It seems like it's still hanging up with trying to load the ep_etherpad-lite plugin. As you mentioned above, the logs should look like the following, but it's hanging after logging the npm version. I mentioned to Ryan above that the logs indicate NPM `version 1.1.37`, but when I check the version via the Console, it reports `version 6.12.0`. Do you think that if the application is really using version 1.1.37 that this could cause a timeout or silent crash or something in Etherpad?

    [2021-02-18 21:38:07.061] [INFO] plugins - npm --version: 6.14.11
    [2021-02-18 21:38:09.580] [INFO] plugins - Loading plugin ep_etherpad-lite...
    [2021-02-18 21:38:09.583] [INFO] plugins - Loaded 1 plugins

@JohnMcLear Progress on the Azure side. I'm loading the correct version of NPM at runtime now (6.12.0), but Etherpad is silently failing after it logs the npm version. I turned on debug logging, but significant showed up in the log files. 

What happens after the npm version is printed? I don't have any plugins installed. Could it be a permissions issue? Does it create folder if it doesn't exist or something like that?

When I run locally, the order of the log entries is different. For example, locally I have "Starting Etherpad..." logged before "npm --version: 6.14.11", however on Azure I see the npm version log, but I do not see "Starting Etherpad...". Similarly, locally I see "Running npm to get a list of installed plugins..." after "Starting Etherpad..." and before "npm --version" is logged, but I don't see "Running npm to get a list of installed plugins..." on the server either.

I assume this is due to asynchronous threads, but is there a clue in this? I'm trying to trace through the code while adding new log entries to find where it's breaking down. I know you're out today, but I'd be open to any tips or hints when you get home! =)

Here's the complete log from a fresh run from Azure this morning.

    [36m[2021-02-24 14:34:49.871] [DEBUG] console - [39mRunning on Node v12.13.0 (minimum required Node version: 10.17.0)
    [36m[2021-02-24 14:34:50.074] [DEBUG] AbsolutePaths - [39mC:\home\site\wwwroot\node_modules\ep_etherpad-lite does not end with "src"
    [32m[2021-02-24 14:34:50.089] [INFO] console - [39mAll relative paths will be interpreted relative to the identified Etherpad base dir: C:\home\site\wwwroot
    [36m[2021-02-24 14:34:50.105] [DEBUG] AbsolutePaths - [39mRelative path "settings.json" can be rewritten to "C:\home\site\wwwroot\settings.json"
    [36m[2021-02-24 14:34:50.105] [DEBUG] AbsolutePaths - [39mRelative path "credentials.json" can be rewritten to "C:\home\site\wwwroot\credentials.json"
    [32m[2021-02-24 14:34:50.136] [INFO] console - [39msettings loaded from: C:\home\site\wwwroot\settings.json
    [32m[2021-02-24 14:34:50.168] [INFO] console - [39mNo credentials file found in C:\home\site\wwwroot\credentials.json. Ignoring.
    [32m[2021-02-24 14:34:50.183] [INFO] console - [39mUsing skin "colibris" in dir: C:\home\site\wwwroot\src\static\skins\colibris
    [36m[2021-02-24 14:34:50.183] [DEBUG] AbsolutePaths - [39mRelative path "./SESSIONKEY.txt" can be rewritten to "C:\home\site\wwwroot\SESSIONKEY.txt"
    [32m[2021-02-24 14:34:50.199] [INFO] console - [39mSession key loaded from: C:\home\site\wwwroot\SESSIONKEY.txt
    [36m[2021-02-24 14:34:50.215] [DEBUG] AbsolutePaths - [39mRelative path "var/dirty.db" can be rewritten to "C:\home\site\wwwroot\var\dirty.db"
    [33m[2021-02-24 14:34:50.230] [WARN] console - [39mDirtyDB is used. This is not recommended for production. File location: C:\home\site\wwwroot\var\dirty.db
    [32m[2021-02-24 14:34:50.246] [INFO] console - [39mRandom string used for versioning assets: 0f458912
    [36m[2021-02-24 14:34:57.184] [DEBUG] runCmd - [39mExecuting command: "C:\Program Files (x86)\npm\6.12.0\npm" --version
    [36m[2021-02-24 14:34:57.200] [DEBUG] runCmd - [39mcwd: C:\home\site\wwwroot
    [36m[2021-02-24 14:34:57.216] [DEBUG] runCmd - [39mPath=C:\Python27;C:\Program Files (x86)\nodejs\12.13.0;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0\;C:\Program Files\Microsoft Network Monitor 3\;C:\Program Files\Git\cmd;C:\Users\imgadmin\AppData\Roaming\npm;C:\Program Files (x86)\nodejs\;C:\Program Files (x86)\Mercurial\;c:\Program Files (x86)\Microsoft ASP.NET\ASP.NET Web Pages\v1.0\;C:\Users\imgadmin\AppData\Local\Microsoft\WindowsApps;;C:\Program Files (x86)\dotnet;C:\Program Files\dotnet;C:\windows\system32\config\systemprofile\AppData\Local\Microsoft\WindowsApps;C:\Program Files\Java\zulu8.46.0.19-jre8.0.252-win_x64\bin;
    [31m[2021-02-24 14:34:57.263] [ERROR] console - [39m(node:3464) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
    [36m[2021-02-24 14:34:59.481] [DEBUG] runCmd - [39mCommand returned successfully: "C:\Program Files (x86)\npm\6.12.0\npm" --version
    [32m[2021-02-24 14:34:59.497] [INFO] plugins - [39mnpm --version: 6.12.0

Progress so far.

server.js -> in Azure, this test at the bottom is failing. I commented it out and just directly run start and it seems to get further. 

    /* 
    if (require.main === module){ 
      logger.warn("Calling server.start()")
      exports.start();
    }
    else{
      logger.warn("Failed test to call server.start()")
    }*/
    
    exports.start();

It's breaking on interpreting the results of `npm ls --long --json --depth=0 --no-production` 

    [36m[2021-02-24 15:51:34.567] [DEBUG] runCmd - [39mAttempting: npm ls --long --json --depth=0 --no-production
    [36m[2021-02-24 15:51:34.567] [DEBUG] runCmd - [39mCommand returned successfully: npm ls --long --json --depth=0 --no-production
    [31m[2021-02-24 15:51:34.599] [ERROR] server - [39mError occurred while starting Etherpad
    [31m[2021-02-24 15:51:34.616] [ERROR] server - [39mTypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be one of type string, Buffer, or URL. Received type undefined
        at Object.realpath (internal/fs/promises.js:457:10)
        at C:\home\site\wwwroot\node_modules\ep_etherpad-lite\static\js\pluginfw\plugins.js:99:30
        at Array.map (<anonymous>)
        at Object.exports.getPackages (C:\home\site\wwwroot\node_modules\ep_etherpad-lite\static\js\pluginfw\plugins.js:94:50)
        at processTicksAndRejections (internal/process/task_queues.js:93:5)
        at async Object.exports.update (C:\home\site\wwwroot\node_modules\ep_etherpad-lite\static\js\pluginfw\plugins.js:67:20)
        at async Object.exports.start (C:\home\site\wwwroot\node_modules\ep_etherpad-lite\node\server.js:146:5)

Jumping into a meeting, but that's my morning progress.

--- a/src/node/utils/Minify.js
+++ b/src/node/utils/Minify.js
@@ -168,7 +168,7 @@ const minify = async (req, res) => {
     if (plugins.plugins[library] && match[3]) {
       const plugin = plugins.plugins[library];
       const pluginPath = plugin.package.realPath;
-      filename = path.relative(ROOT_DIR, path.join(pluginPath, libraryPath));
+      filename = path.join(pluginPath, libraryPath);
       // On Windows, path.relative converts forward slashes to backslashes. Convert them back
       // because some of the code below assumes forward slashes. Node.js treats both the backlash
       // and the forward slash characters as pathname component separators on Windows so this does
@@ -241,7 +241,7 @@ const statFile = async (filename, dirStatLimit) => {
   } else {
     let stats;
     try {
-      stats = await fs.stat(path.join(ROOT_DIR, filename));
+      stats = await fs.stat(path.resolve(ROOT_DIR, filename));
     } catch (err) {
       if (err.code === 'ENOENT') {
         // Stat the directory instead.
@@ -329,7 +329,7 @@ const getFileCompressed = async (filename, contentType) => {
 
 const getFile = async (filename) => {
   if (filename === 'js/require-kernel.js') return requireDefinition();
-  return await fs.readFile(path.join(ROOT_DIR, filename));
+  return await fs.readFile(path.resolve(ROOT_DIR, filename));
 };
 
 exports.minify = (req, res, next) => minify(req, res).catch((err) => next(err || new Error(err)));

--- a/src/node/utils/MinifyWorker.js
+++ b/src/node/utils/MinifyWorker.js
@@ -12,7 +12,7 @@ const compressJS = (content) => Terser.minify(content);
 
 const compressCSS = (filename, ROOT_DIR) => new Promise((res, rej) => {
   try {
-    const absPath = path.join(ROOT_DIR, filename);
+    const absPath = path.resolve(ROOT_DIR, filename);
 
     /*
        * Changes done to migrate CleanCSS 3.x -> 4.x:

